src/org/labkey/targetedms/parser/speclib/BlibSpectrumReader.java

```diff
@@ -183,7 +147,7 @@ public static Path redundantBlibPath(@Nullable Path blibPath)
             StringBuilder sql = new StringBuilder("SELECT rt.retentionTime, rt.bestSpectrum, rs.peptideModSeq, rs.precursorCharge, ssf.fileName from RetentionTimes AS rt ");
             sql.append(" INNER JOIN RefSpectra AS rs ON (rt.RefSpectraID = rs.id)");
             sql.append(" INNER JOIN SpectrumSourceFiles ssf ON (rt.SpectrumSourceID = ssf.id)");
-            sql.append(" WHERE rs.peptideModSeq='"+blibPeptide+"'");
+            sql.append(" WHERE rs.peptideModSeq='").append(blibPeptide).append("'");
```

Comment:
This has the potential for SQL injection. Is blibPeptide guaranteed to be safe to concatenate here? If not, you should use a JDBC parameter.

Reply:
This is not a user provided string the way this method is called right now.  But that may change later, so I have updated it to use a PreparedStatement instead.  Thanks for pointing it out!